### ISSUE TITLE
philosophy

### ISSUE THREAD

**Issue by mattdesl** [stackgl/webglew#2]
I am beginning to question the use of this module for a couple reasons:
1. Prefixing is deprecated and no longer a thing in modern browsers (unless I'm mistaken)
2. Calling `webglew(gl)` will enable _all_ extensions on that GL context
   - does not respect principle of least astonishment
   - causes obscure bugs depending on program order: i.e. using `gl-shader` before `gl-texture2d` can lead to shader errors, but not the other way around
   - enabling an obscure (and unused) OES extension was causing our application to crash on some crappy Android GPUs
3. WeakMap adds some overhead to the bundle

I feel like `gl.getExtension` is probably fine nowadays, no?

``` js
var ext = gl.getExtension('OES_texture_float')
if (ext) {
    ...
}
```

**Comment by mikolalysenko** [comment]
Yeah, I think so.  Part of the reason for using it initially was to avoid loading extensions twice and also to deal with the hassle of different prefixes (which as you point out is not even a thing anymore).

Maybe we should move to deprecate it?

**Comment by mattdesl** [comment]
Does multiple `gl.getExtension` calls have any negative impact? If so we could still wrap it in a weak map for caching, but with an api like `webglew(gl, 'OES_texture_float')`

**Comment by mikolalysenko** [comment]
I'm not really sure if it matters, though it would be good to test.  I could definitely see this being browser/extension dependent though so it makes me a bit nervous.

**Comment by mattdesl** [comment]
Skimming the spec, it seems to go through the usual motions of checking context loss on each call, raising `INVALID_VALUE` where necessary, etc. This might actually be a good thing since the cached value is no longer usable if the context has been lost and restored. 

Also: 

> Multiple calls to getExtension with the same extension string, taking into account case-insensitive comparison, must return the same object as long as the extension is enabled.

But yeah, worth testing.

**Comment by mikolalysenko** [comment]
Ok, then that settles it.  I move to deprecate this module going forward.
